archinstall/lib/user_interaction.py

```diff
@@ -206,40 +242,69 @@ def ask_for_main_filesystem_format():
 		'f2fs' : 'f2fs'
 	}
 
-	value = generic_select(options.values(), "Select which filesystem your main partition should use (by number or name): ")
+	value = generic_select(options, "Select which filesystem your main partition should use (by number or name): ",
+						  allow_empty_input=False)
 	return next((key for key, val in options.items() if val == value), None)
 
-def generic_select(options, input_text="Select one of the above by index or absolute value: ", sort=True):
+def generic_select(options, input_text="Select one of the above by index or absolute value: ", allow_empty_input=True, options_output=True):
 	"""
 	A generic select function that does not output anything
 	other than the options and their indexes. As an example:
 
 	generic_select(["first", "second", "third option"])
-	1: first
-	2: second
-	3: third option
+	0: first
+	1: second
+	2: third option
+
+	When the user has entered the option correctly,
+	this function returns an item from list, a string, or None
 	"""
 
-	if type(options) == dict: options = list(options)
-	if sort: options = sorted(list(options))
-	if len(options) <= 0: raise RequirementError('generic_select() requires at least one option to operate.')
-
-	for index, option in enumerate(options):
-		print(f"{index}: {option}")
-
-	selected_option = input(input_text)
-	if len(selected_option.strip()) <= 0:
-		return None
-	elif selected_option.isdigit():
-		selected_option = int(selected_option)
-		if selected_option > len(options):
-			raise RequirementError(f'Selected option "{selected_option}" is out of range')
-		selected_option = options[selected_option]
-	elif selected_option in options:
-		pass # We gave a correct absolute value
-	else:
-		raise RequirementError(f'Selected option "{selected_option}" does not exist in available options: {options}')
+	# Checking if options are different from `list` or `dict`
+	if type(options) not in [list, dict]:
+		log(" * It looks like there are something wrong with provided options. Maybe it's time to open an issue on GitHub! * ", fg='red')
+		log(" * Here are the link: https://github.com/archlinux/archinstall/issues * ", fg='yellow')
+		raise RequirementError("generic_select() requires list or dictionary as options.")
+	if type(options) == dict: options = sorted(list(options.values()))  # To allow only `list` and `dict`, converting values of options and sorting them here. Therefore, now we can only provide the dictionary itself
```

Comment:
We have to be careful calling `sorted()`, this might mess up the magic mentioned above.